app/scripts/components/criteria/variables.service.js

```diff
@@ -31,6 +33,11 @@ angular.module("chuvApp.components.criteria").factory("Variable", [
     };
 
     resource.hierarchy = function() {
+      var hierarchy = cache.get("hierarchy");
+      if (!angular.isUndefined(hierarchy)) {
+        return $q.resolve(hierarchy);
```

Comment:
here we used to have Promise.resolve. Unlike $q.resolve, Promise.resolve does not triggers a digest cycle, so updates with data taken from cache was not refreshed in the view and had to wait for a digest cycle triggered elsewhere. I think that is what gave the impression that the cache is sluggish.